cs-minibot-platform-src/src/main/java/examples/gobot/GoBot.java

```diff
@@ -0,0 +1,375 @@
+                sum+= lapTimes.get(i) - this.startTime;
+            } else
+            sum += (lapTimes.get(i) - lapTimes.get(i-1));
+        }
+        return sum/1000000000;
+    }
+
+    public void setBotState(int state) {
+        this.botState = state;
+    }
+
+    private void printState() {
+        System.out.printf("HUMAN: %s, AI: %s", humanTime, botTime);
+    }
+
+    @Override
+    public void run() {
+        //this.setTimer(System.nanoTime());
+        this.fwm = (FourWheelMovement)
+                BaseStation
+                        .getInstance().getBotManager().getAllTrackedBots()
+                        .iterator().next().getCommandCenter();
+
+        while (true) {
+            if (lastBotState != botState) {
+                this.startTime = System.nanoTime();
+            }
+            lastBotState = botState;
+            if (botState == HUMAN_PLAYING) {
+                if (lapsDone >= numLaps) {
+                    // Finished!
+                    botState = WAITING;
+                    humanTime = totalTime();
+                    lapTimes.clear();
+                    lapsDone = 0;
+                    printState();
+                } else {
+                    List<VisionObject> vl =  BaseStation.getInstance().getVisionManager()
+                            .getAllLocationData();
+                    if (vl.size() != 0) {
+                        VisionCoordinate vc = vl.get(0).coord;
+                        if (this.course.isInsideTrack(vc)) {
+                            inTrack = true;
+                            if (course.getStartArea().contains(vc.x, vc.y)) {
+                                if (!crossedLapLine && !reachedMiddle) {
+                                    this.crossedLapLine = true;
+                                } else if (crossedLapLine && reachedMiddle) {
+                                    this.lapsDone++;
+                                    this.reachedMiddle = false;
+                                    this.lapTimes.add(System.nanoTime());
+                                }
+                            } else if (course.getMiddleArea().contains(vc.x, vc.y)) {
+                                if (!reachedMiddle) {
+                                    this.reachedMiddle = true;
+                                }
+                            }
+                        } else {
+                            System.out.println("Out of bounds");
+                            inTrack = false;
+                        }
+                    }
+                }
+            }
+            else if (botState == BOT_PLAYING) { //ASSUMING CCL TRACK
+                if (lapsDone >= numLaps) {
+                    // Finished!
+                    botState = WAITING;
+                    botTime = totalTime();
+                    lapTimes.clear();
+                    lapsDone = 0;
+                    printState();
+                }
+                else if (aiType == 0){
```

Comment:
make a 
public static final int AITYPE_BASIC or something to make it clear what this type means